pinot-core/src/main/java/com/linkedin/pinot/core/operator/aggregation/groupby/AggregationGroupByOperator.java

```diff
@@ -60,7 +60,8 @@ public AggregationGroupByOperator(List<AggregationInfo> aggregationsInfoList, Gr
     Preconditions.checkNotNull(groupBy);
     Preconditions.checkNotNull(projectionOperator);
 
-    _groupByExecutor = new DefaultGroupByExecutor(projectionOperator, aggregationsInfoList, groupBy, numGroupsLimit);
+    _groupByExecutor =
+        new DefaultGroupByExecutor(projectionOperator.getDataFetcher(), aggregationsInfoList, groupBy, numGroupsLimit);
```

Comment:
GroupKeyGenerator needs global info (non block specific) to compute stuff like max cardinality product. I was debating over providing those api's on block vs just passing dataFetcher that already has those methods. Based on our discussions, I will explore not exposing dataFetcher outside of projection.